src/arch/xtensa/CMakeLists.txt

```diff
@@ -110,6 +110,10 @@ endif()
 # C & ASM flags
 target_compile_options(sof_options INTERFACE ${stdlib_flag} -fno-inline-functions ${XTENSA_C_ASM_FLAGS})
 
+if(SPARSE STREQUAL "y")
+	target_compile_options(sof_options INTERFACE -D__CHECKER__ -Wno-attributes)
```

Comment:
`__CHECKER__` is really vague, anything more specific would be better, off the cuff: `SPARSE_STATIC_CHECK`.  The longer the better, makes people think twice before using it :-)

@joshtriplett, has sparse any built-in macro for this?

Reply:
As far as I know, `__CHECKER__` is the macro to identify Sparse. The maintainer of Sparse might be willing to add a more specific macro, perhaps `__SPARSE_CHECKER__`, if you send a patch.

Reply:
> As far as I know, __CHECKER__ is the macro to identify Sparse. 

Thx @joshtriplett ! Interesting, I did not expect that.

In that case @lyakh why the `-D__CHECKER__` here?

Reply:
Good hearing from you again @joshtriplett !
@lyakh can you also align with @ceolin and @andyross as we should upstream SPARSE into Zephyr directly and reuse it as a module client.

Reply:
I'd love to see a SPARSE integration for Zephyr (just so long as everyone else fixes all the resulting false-ish positives, heh).   We get coverity output semi-regularly, so the tree should hopefully be relatively well prepared.

Reply:
@marc-hb no, the kernel uses `__CHECKER__`, that's why I took it too

Reply:
Dunno why the kernel does this but I think you can remove `-D__CHECKER__` here based on what @joshtriplett wrote and this: https://github.com/lucvoo/sparse/blob/cf8f104749f5bca/predefine.c#L84

Reply:
@marc-hb I further checked whether `__CHECKER__` is defined automatically by sparse. Indeed it is if you just call `sparse` directly. But if you use the cgcc sparse compiler wrapper, then it isn't. And we do use cgcc to do all the preprocessing.

Reply:
OK, redefining something that spare already defines is quite subtle and threw me off (and the name even more), please add a one-line comment about that.